week1/utils.go

```diff
@@ -0,0 +1,73 @@
+package main
+
+import (
+	"fmt"
+	"strings"
+	"strconv"
+)
+
+func add(x, y float64) float64 {
+	return x + y
+}
+
+func subtract(x, y float64) float64 {
+	return x - y
+}
+
+func multiply(x, y float64) float64 {
+	return x * y
+}
+
+func divide(x, y float64) (result float64, err error) {
+	if y == 0 {
+		err = fmt.Errorf("divide by zero")
+		return
+	}
+
+	result = float64(x) / float64(y)
+	return
+}
+
+func parseInput(input string) (x float64 , y float64 , op string, err error) {
+	s := strings.Fields(input)
+	op = s[1]
```

Comment:
better check length of `s` first before access it to avoid `index out of range` runtime error

otherwise LGTM.